packages/office-ui-fabric-react/src/components/ResizeGroup/ResizeGroup.base.tsx

```diff
@@ -314,15 +320,65 @@ export class ResizeGroupBase extends BaseComponent<IResizeGroupProps, IResizeGro
   // for the initial render.
   private _hasRenderedContent = false;
 
+  public static getDerivedStateFromProps(props: IResizeGroupProps, state: IResizeGroupState) {
+    if (state.prevProps !== props) {
+      const { data: prevData } = state.prevProps!; // We set initial prevProps in getInitialState.
+      const { data, onBatchScaleData } = props;
+      const { renderedData } = state;
+
+      // If cache key is defined and equal to cache key of previous props' data we can do an optimization
+      if (prevData.cacheKey && data.cacheKey === prevData.cacheKey) {
```

Comment:
What happens if this prop update also changed the scaling steps? Say a new scaling step was added or one was removed. The scalingStepCount would be incorrect

Reply:
This will definitely create bugs if scaling steps are added or removed. However, in my opinion, that should be rare scenario and we can add it as a caveat to using the batch scaling of data.

Reply:
I don't think it's something that rare in the grand scheme of things (especially for a shared component like this), think of add-ins being inserted at runtime or if we enable user customization of the Ribbon.

I'm in favor of improving performance, but not at the expense of correctness. This will be hard to debug when it goes wrong in the future.

Reply:
Rather than making things in the shared component more complex, is it possible to keep the existing API but add these smarts in the Ribbon's implementation of onGrowData  onReduceData?

Reply:
So, when we batchScaleData, at least in the current implmentation, I check to verify that the final cacheKey after batchScaling is the same before rendering. If the scalingSteps change, I would expect the cacheKey to change. If it does not, it should be ok to still render, because the current rendered data has the same key.